R/calcFeedBaskets.R

```diff
@@ -35,12 +35,28 @@ calcFeedBaskets <- function(non_eaten_food = FALSE, # nolint
                                     start_year = 2010, end_year = 2050, type = "s")
 
     if (fadeout) {
-      out <- calibDECLINE2050
+      calibChosen <- calibDECLINE2050
     } else {
-      out <- calibCONST
+      calibChosen <- calibCONST
     }
 
-    out <- out + feedBasketsUNCALIB
+    out <- calibChosen + feedBasketsUNCALIB
+
+    # do not allow the calibration to introduce the inconsistency of
+    # more livestock products in feed basket than produced;
+    # apply a reduction factor for calibration to avoid this
+    milkInpDairy <- list(sys = "sys_dairy", kall = "livst_milk") # item showing milk input for dairy production
+    if (any(feedBasketsUNCALIB[, , milkInpDairy] >= 1)) {
+      stop("more livestock products in feed basket than being produced already for uncalibrated feed baskets")
+    }
+    calibReductionFact <- (1 - feedBasketsUNCALIB[, , milkInpDairy]) /
+      calibChosen[, , milkInpDairy] - .Machine$double.eps
+    calibReductionFact[(calibReductionFact >= 1) | (calibReductionFact <= 0)] <- 1
+    # note that calibReductionFact can only be smaller than 1, if the positive difference
+    # "feedBasketsUNCALIB[,,milkInpDairy] - 1" is smaller than the positive "calibChosen[,,milkInpDairy]"
```

Comment:
I assume that feedBasketsUNCALIB[,,milkInpDairy] is always positive yes (from a content perspective). Otherwise there would be a feed output from producing animal products, which sounds strange to me.

Reply:
Maybe the above explanation is better.

Reply:
I am sorry, you are of course right. I meant: Is "calibChosen[,,milkInpDairy]" always positive?

Reply:
In general no, but in case calibReductionFact is smaller than 1 yes.

Reply:
If calibChosen[,,milkInpDairy] is negative then the only way calibReductionFactor can be >0 (based on calculation in line 52) is that (1 - feedBasketsUNCALIB[, , milkInpDairy]) < 0. But this is ruled out by the check in line 50. However calibReductionFactor  > 0 is a condition for calibReductionFactor  < 1 at line 55, based on the line above

Reply:
So if calibReductionFactor < 1 at the time of the comment calibChosen[,,milkInpDairy]  was negative

Reply:
The message i wanted to convey is that the reduction is only applied if there is a positive calibration term.
But now that I think about it, this information is not important. It just causes confusion and over-complicates things.